internal/cmd/kafka/command_topic_consume.go

```diff
@@ -70,17 +66,33 @@ func newConsumeCommand(prerunner pcmd.PreRunner, clientId string) *cobra.Command
 	return cmd
 }
 
-func (c *hasAPIKeyTopicCommand) consume(cmd *cobra.Command, args []string) error {
+func (c *command) consume(cmd *cobra.Command, args []string) error {
 	topic := args[0]
 
-	valueFormat, err := cmd.Flags().GetString("value-format")
+	cluster, err := c.Config.Context().GetKafkaClusterForCommand()
 	if err != nil {
 		return err
 	}
+	if cluster.APIKey == "" {
+		apiKey, err := cmd.Flags().GetString("api-key")
+		if err != nil {
+			return err
+		}
 
-	cluster, err := c.Config.Context().GetKafkaClusterForCommand()
-	if err != nil {
-		return err
+		apiSecret, err := cmd.Flags().GetString("api-secret")
+		if err != nil {
+			return err
+		}
+
+		if apiKey != "" && apiSecret != "" {
+			cluster.APIKey = apiKey
+			cluster.APIKeys[cluster.APIKey] = &v1.APIKeyPair{
+				Key:    apiKey,
+				Secret: apiSecret,
+			}
+		} else {
+			return &errors.UnspecifiedAPIKeyError{ClusterID: cluster.ID}
+		}
```

Comment:
small suggestion: this block seems to be repeated for produce and consume files; perhaps it can go in a helper function in command_topic.go